packages/appium/docs/en/ecosystem/build-drivers.md

```diff
@@ -2,4 +2,617 @@
+
+Your `package.json` must contain an `appium` field, like this (we call this the 'Appium manifest'):
+
+    ```json
+    {
+        ...,
+        "appium": {
+            "driverName": "fake",
+            "automationName": "Fake",
+            "platformNames": [
+              "Fake"
+            ],
+            "mainClass": "FakeDriver"
+        },
+        ...
+    }
+    ```
+
+The required subfields are:
+
+* `driverName`: this should be a short name for your driver.
+* `automationName`: this should be the string users will use for their `appium:automationName`
+capability to tell Appium to use *your* driver.
+* `platformNames`: this is an array of platform names considered valid for your driver. When a user
+sends in the `platformName` capability to start a session, it must be included in this list for
+your driver to handle the session.
+* `mainClass`: this is a named export from your `main` field. It must be a class which extends
+Appium's `BaseDriver` (see below).
+
+### Extend Appium's `BaseDriver` class
+
+Ultimately, your driver is much easier to write because most of the hard work of implementing the
+WebDriver protocol and handling certain common logic is taken care of already by Appium. This is
+all encoded up as a class which Appium exports for you to use, called `BaseDriver`. It is exported
+from `appium/driver`, so you can use one of these styles to import it and create your *own* class
+that extends it:
+
+```js
+import {BaseDriver} from 'appium/driver';
+// or: const {BaseDriver} = require('appium/driver');
+
+export class MyDriver extends BaseDriver {
+}
+```
+
+### Make your driver available
+
+That's basically it! With a Node.js package exporting a driver class and with a correct Appium
+manifest, you've got yourself an Appium driver! Now it doesn't *do* anything, but you can load it
+up in Appium, start and stop sessions with it, etc...
+
+To make it available to users, you could publish it via NPM. When you do so, your driver will be
+isntallable via the Appium CLI:
+
+```
+appium driver install --source=npm <driver-package-on-npm>
+```
+
+### Developing your driver
+
+How you develop your driver is up to you. It is convenient, however, to run it from within Appium
+without having to do lots of publishing and installing. The most straightforward way to do this is
+to include the most recent version of Appium as a `devDependency`, and then also your own driver,
+like this:
+
+```json
+{
+    "devDependencies": {
+        ...,
+        "appium": "^2.0.0",
+        "your-driver": "file:.",
+        ...
+    }
+}
+```
+
+Now, you can run Appium locally (`npx appium`), and because your driver is listed as a dependency
+alongside it, it will be automatically "installed" and available. You can design your e2e tests
+this way, or if you're writing them in Node.js, you can simply import Appium's start server methods
+to handle starting and stopping the Appium server in Node.
```

Comment:
We will probably want to reference the basedriver E2E suite function as well.  This is being moved into `appium/test` in #16907